shell.nix

```diff
@@ -0,0 +1,53 @@
+{ pkgs ? import <nixpkgs> {} }:
+	let
+	overrides = (builtins.fromTOML (builtins.readFile ./rust-toolchain.toml));
+	libPath = with pkgs; lib.makeLibraryPath [
+		# load external libraries that you need in your rust project here
+	];
+in
+pkgs.mkShell rec {
+	# Kanidm dependencies
+	buildInputs = with pkgs; [
+		pkg-config
+		
+		git
+		rustup
```

Comment:
Why? Either use user's preinstalled rustup and don't do anything here, or add rustc and cargo here to use the compiler from nixpkgs. I'd do the latter because that way everything is managed by Nix.

Those two variants also don't require you to mess with PATH like that.

Reply:
Rustup is the idiomatic way to work with Rust in nixos. We are able to include RustC & cargo at a specific version using nix but that removes control for the user to be able to run a different toolchain. I thought it best to stick with the standard and better documented case of just using rustup. Happy to do more investigation though if we think that it will be cleaner in the end doing it ourselves.

Reply:
I don't know, usually I see people use either nixpkgs rustc, or fenix/rust-overlay. All three of these are options that "just work" (just need to run nix develop/nix-shell), and only place files in the Nix store.

Mostly it feels wrong to me to use Nix to just install an updater program instead of getting the real compiler straight away, I suppose.

If someone does want to use a different version, they can always comment it out here locally or run another inner nix shell or whatever.

Reply:
Changed to cargo & rustc